src/python/pants/jvm/jdk_rules.py

```diff
@@ -71,6 +71,11 @@ def immutable_input_digests(self) -> dict[str, Digest]:
         return {**self.coursier.immutable_input_digests, self.bin_dir: self._digest}
 
 
+@dataclass
+class JdkSetup:
+    jdk: JdkEnvironment
```

Comment:
> `JdkSetup` now becomes `JdkEnvironment` (constructed by `JdkRequest`), and the new `JdkSetup` global object is just a simple wrapper for `JdkEnvironment`

I'm not seeing the reason for this split. Is the idea that a `JdkSetup` is going to end up with more fields, all of which callers will need in order to use a `Jdk` (and thus, they're all requesting a `JdkSetup` rather than the `JdkEnvironment`)?

Reply:
This is more indirection, written with the intention of _removing_  -- currently we have a `JdkSetup` which is specifically to construct a "global" `JdkEnvironment`. In the followup, `JdkSetup` will go away, leaving just `JdkEnvironment`, which will not be injected, but will be requested based on the JDK version in the requests.